### PR TITLE
docs: update runpath/rpath docs

### PR DESCRIPTION
**Pull request by jsquyres**
Add a bunch of clarifications / explanations of Open MPI's
runpath/rpath behavior.

Signed-off-by: Jeff Squyres <jsquyres@cisco.com>

Quick link to see the changes of this PR rendered on RTD: https://ompi--10624.org.readthedocs.build/en/10624/installing-open-mpi/configure-cli-options/installation.html

### CODE REVIEW

**Comment by jsquyres** [comment]
@bwbarrett and I talked about this, and we *think* it's right.  So I'm going to take this out of draft.

@bwbarrett is going to do a closer review and see if anything pops out at him.

**Comment by jsquyres** [comment]
Added a link to some preview slides for a Linux course on shared libraries + the linker with explicit permission from the author.

File: docs/installing-open-mpi/configure-cli-options/installation.rst
Comment (bwbarrett) [inline docs/installing-open-mpi/configure-cli-options/installation.rst:57]: > "are only relevant to the flags the wrapper compilers add when compiling user applications."

The wrapper compilers themselves are still built with whatever flags OMPI was built with.

File: docs/installing-open-mpi/configure-cli-options/installation.rst
Comment (jsquyres) [reply docs/installing-open-mpi/configure-cli-options/installation.rst]: We talked about this verbally.  I just pushed a new commit that separates the text of how the wrappers use rpath/runpath from rpath/runpath of Open MPI itself.  See:

* https://ompi--10624.org.readthedocs.build/en/10624/installing-open-mpi/configure-cli-options/installation.html
* https://ompi--10624.org.readthedocs.build/en/10624/installing-open-mpi/configure-cli-options/rpath-and-runpath.html

**Review by bwbarrett — APPROVED** [review]
Works for me with the squash.

**Comment by jsquyres** [comment]
Squashed.  Will merge when CI completes.

File: docs/installing-open-mpi/configure-cli-options/installation.rst
Comment (haampie) [inline docs/installing-open-mpi/configure-cli-options/installation.rst:48]: This is incorrect. `--disable-wrapper-runpath` will drop `-Wl,--enable-new-dtags` and leave it up to the compiler defaults (which is runpath for ages already in GCC).

https://github.com/open-mpi/ompi/issues/10560

**Comment by haampie** [comment]
@bwbarrett can you have another look here?

File: docs/installing-open-mpi/configure-cli-options/installation.rst
Comment (jsquyres) [reply docs/installing-open-mpi/configure-cli-options/installation.rst:48]: Gah; you're right.  Thanks for catching this.  Reuti also offered some suggestions in https://www.mail-archive.com/users@lists.open-mpi.org/msg34968.html; I'll make a follow-on PR.

Comment (haampie) [reply docs/installing-open-mpi/configure-cli-options/installation.rst:48]: Could the follow-up PR be a fix that makes `--enable-wrapper-runpath` set `-Wl,--enable-new-dtags` and `--enable-wrapper-rpath` -> `-Wl,--disable-new-dtags`? To me it's a bug in openmpi that explicitly asking to enable rpath and disable runpath leaves you with linker defaults, which is runpath in almost all cases

Comment (jsquyres) [reply docs/installing-open-mpi/configure-cli-options/installation.rst:48]: That's a fair point.

I made the follow-on PR #10652 -- please have a look.

I did *not* change `configure` or the wrappers behavior yet -- #10652 is about documenting the current behavior.  We should follow up on #10560 to decide on exactly how to change the behavior, and then have a new PR to both change the behavior and update the docs to reflect the new behavior.

**Pull request merged**
